### ISSUE TITLE
ModKit Request

### ISSUE THREAD

**Issue by Phlo123** [StrangeLoopGames/EcoIssues#2395]
Request for ModKit.

Access to adding the following items/objects:

Tools 
AvatarParts; ClothingItem
Block; WorldBlock
Skill Icon
Menu Window
PlantBlock, (PlantInteractionBlock?)
Edit: AnimalEntity or Object; And the ability to run multiple unity3d files on a server.

Note:
Some of the items in the list above require time investment from me as well to understand how they function, and I am dedicating about +10h each day in learning new things about unity and 3d modeling.
Even tho my skills are very limited right now, I am learning. With time I will get better, and I wont have to bother the devs with every issue I run in to.
I don't want you to think that investing effort and time in the modkit will be in vain, and when i have the knowledge to sprint trough Unity I can help new players that will want to use the modkit. I already have all the help I need with code ,and programming in general, from Clay and Emmeck and I have a very high understanding of Photoshop, Illustrator, CorelDraw & CorelPaint; as I have worked with them for 8-9 years.
The bottom line is: I am focused on Eco and modding the game, and I want to learn more!

**Comment by Phlo123** [comment]
Update on the request

**Comment by Phlo123** [comment]
Update on the quides request:
- How to make a light source have Curved Shader (or equivalent)?
- ClothingItem guide.
- WorldObject with RandomStateMachine animation doesn't work. Maybe you are using another way to call for random animations for objects.
- How to add new interface buttons to the game.

**Comment by johnkslg** [comment]
We should get some more mod support in for beta.

**Comment by Kodekc** [comment]
Need to fix the icon that is used in the description and in the chat
![1](https://user-images.githubusercontent.com/16691965/41862241-fc2777e2-78ab-11e8-8f0a-9f28e3589665.PNG)
![2](https://user-images.githubusercontent.com/16691965/41862242-fc56274a-78ab-11e8-974b-7ef2e3f885c7.PNG)

hash/commit ModKit: [0749fba](https://github.com/StrangeLoopGames/EcoModKit/commit/8d4be15f3a78f670d75ba0780831294f8b0d00c4)

**Comment by metachronism** [comment]
most of these have been added.  Icons issue has another bug
